library/summary.mli

```diff
@@ -11,17 +11,12 @@
 (** This module registers the declaration of global tables, which will be kept
    in synchronization during the various backtracks of the system. *)
 
-type marshallable =
-  [ `Yes       (* Full data will be marshalled to disk                        *)
-  | `No        (* Full data will be store in memory, e.g. for Undo            *)
-  | `Shallow ] (* Only part of the data will be marshalled to a slave process *)
-
```

Comment:
You are removing documentation and IMO not properly explaining the semantics of the new boolean flag. Please improve.

Reply:
Funnily enough, the boolean was already documented a few lines below. Probably a left there from before the commit which introduced the `Shallow` logic that this PR removes. What do you want me to improve?

Reply:
IMHO the below comment as well as this one it does explain nothing. Ok, mashalled to disk, what's the difference?